src/country_details.jsx

```diff
@@ -0,0 +1,102 @@
+import { useLocation } from "react-router-dom";
+import { useEffect, useState } from "react";
+import { createSearchParams, useSearchParams } from "react-router-dom";
+import { useParams } from 'react-router-dom'
+import get from './utils/axios-configure'
+import axios from 'axios'
+import { margin, style, width } from "@mui/system";
+import TinyFlag from "tiny-flag";
+import {url_flag,url_country} from "./utils/const"
+
+
+
+
+
+export default function Details() {
+    const [searchParams, setSearchParams] = useSearchParams();
+    // const getcode=searchParams.get("code");
+    const queryParams = new URLSearchParams(window.location.search);
+    const { code } = useParams();
+    const [capitals, setCapital] = useState([]);
+    const [phonecodes, setPhonecode] = useState([]);
+    const [currencys, setCurrency] = useState([]);
+    const [flags, setFlag] = useState([]);
+
+    const header = {
+        'Content-Type': 'application/json'
+    };
+
+
+    useEffect(() => {
+        async function question() {

```

Comment:
because there was an issue when i  call promise .all so i needed to make it async function i think because the calling for the api need to finish resolving all of them in the same time   so that can fetch all the api data without waiting for another api to finish it job that what promise .all do